test/test_primitives/assert_helper.h

```diff
@@ -64,6 +95,67 @@ inline void assert_intersect_primitive_hit(const Collection<T> &collection, cons
   CHECK(rayhit.normal[2] == Approx(static_cast<T>(normal[2])));
 }
 
+
+/**
+  * Used for trimesh, if more than one triangle is used
+  */
+template<typename T, template<typename> typename Collection>
+inline void assert_traverse_bvh_hit_trimesh(const Collection<T> &collection, const Ray<T> &ray, const bool true_hit,
+                                            const T distance, const Vec3r<T> &normal) {
```

Comment:
replace trimesh with collection, because this should work for all collections

Reply:
done